widgets/linechart/internal/axes/value.go

```diff
@@ -72,14 +84,24 @@ func (v *Value) Text() string {
 	if v.text != "" {
 		return v.text
 	}
-	if math.Ceil(v.Rounded) == v.Rounded {
-		return fmt.Sprintf("%.0f", v.Rounded)
+
+	if v.Formatter != nil {
+		return v.Formatter(v.Value)
+	}
+
+	return v.defaultFormatter(v.Rounded)
+}
+
+func (v *Value) defaultFormatter(value float64) string {
```

Comment:
(nit) Looks like this doesn't need to be a method anymore (it doesn't use the receiver), can it be a plain function?

```go
func defaultFormatter(value float64) string { ... }
```

Reply:
Just to understand and learn, what's the benefit (or drawbacks) in these cases, converting a method to a function? I mean, for example, when I have these cases I maintain as an internal method to give the semantic meaning that this function is used internally by the struct and anything else.

Reply:
Happy to explain my position more and please feel free to ask about any comments if in doubt. It is indeed good to mark elements as internal for the reasons you explained. However a plain function can also be internal (unexported, start with lowercase letter), so this shouldn't affect the choice of function Vs. method.

Making function into a method extends the surface of its API by giving it an additional input or output; the receiver (v *Value). A reader of the code now assumes that the method either needs to read the object's state, or it might be modifying the object state as part of what it does. By making something that doesn't need to be a method a plain function, we are making the reader's life easier by being more explicit.

The argument I often make is about using the lowest primitive that does the job. One would not use a variable where a constant is enough, a function where a variable is enough or a method/object where a plain function is enough.

Of course there are languages that like to use objects as a way to package common elements together (like Java). Go's style tends to prefer clean object design as described above.

Reply:
Understood, in those cases to let the user know that the struct will not be used I usually do this change `func (s SomeStruct) someMethod()` or ` func (s *SomeStruct) someMethod()` to `func (SomeStruct) someMethod()`. Although setting the plain func is more visual. Anyway, more than happy to set `defaultFormatter` as a plain function.

Thanks for the explanation!